src/lib/flight_tasks/tasks/Auto/FlightTaskAuto.cpp

```diff
@@ -256,6 +256,26 @@ bool FlightTaskAuto::_evaluateTriplets()
 		_ext_yaw_handler->deactivate();
 	}
 
+	// Calculate the current vehicle state and check if it has updated.
+	State previous_state = _current_state;
+	_current_state = _getCurrentState();
+
+	if (triplet_update || (_current_state != previous_state) || _current_state == State::offtrack) {
+		_updateInternalWaypoints();
+		_mission_gear = _sub_triplet_setpoint.get().current.landing_gear;
+		_yaw_lock = false;
```

Comment:
This is the only line of the hunk that is not copy-paste

The rest comes from below because the execution order did not allow adjusting the heading based on the newly updated waypoints from this loop iteration.